politeiad/backendv2/tstorebe/plugins/usermd/usermd.go

```diff
@@ -84,21 +86,153 @@ func (p *usermdPlugin) Hook(h plugins.HookT, payload string) error {
 // Fsck performs a plugin file system check. The plugin is provided with the
 // tokens for all records in the backend.
 //
+// It verifies the user cache using the following process:
+//
+// 1. For each record, get the user metadata file from the db.
+// 2. Get the user cache for the record's author.
+// 3. Verify that the record is listed in the user cache under the
+//    correct category.  If the record is not found in the user
+//    cache, add it.  The tokens listed in the user cache are
+//    ordered by the timestamp of their most recent status change
+//    from newest to oldest.
+//
 // This function satisfies the plugins PluginClient interface.
 func (p *usermdPlugin) Fsck(tokens [][]byte) error {
 	log.Tracef("usermd Fsck")
 
-	// Verify the user cache using the following process:
-	//
-	// 1. For each record, get the user metadata file from the db.
-	//
-	// 2. Get the user cache for the record's author.
-	//
-	// 3. Verify that the record is listed in the user cache under the
-	//    correct category. The tokens listed in the user cache MUST be
-	//    ordered by the timestamp of their most recent status change
-	//    from newest to oldest. If the record is not found in the user
-	//    cache, add it.
+	// This map holds all fetched records
+	rs := make(map[string]*backend.Record, len(tokens))
```

Comment:
Reading every record into memory is a non-starter.  What happens if there's 10,000 records?